### ISSUE TITLE
Cannot find compilation library location for package 'Microsoft.NETCore.App'

### ISSUE THREAD

**Issue by hannespreishuber** [aspnet/AspNetCore#7450]
Deploying a ASP.NET Core 2.2 Project to IIS 10  with Visual Studio 2017 (an 2019) via FTP 
-works: No Pages Folder on Webs server

Adding a  in visula Studio Project a Razor Page without a model class - deploy just this page 

open page in Browser Error
--Cannot find compilation library location for package 'Microsoft.NETCore.App'

Startup.cs contains 
   services.AddMvc().AddRazorOptions(options => options.AllowRecompilingViewsOnFileChange = true);
(should be Default?)

csproj
```xml
<Project Sdk="Microsoft.NET.Sdk.Web">
  <PropertyGroup>
    <TargetFramework>netcoreapp2.2</TargetFramework>
    <UserSecretsId>7c38072d-0407-48bb-8abc-96cc1d96ac2f</UserSecretsId>
    <TypeScriptToolsVersion>3.1</TypeScriptToolsVersion>
<MvcRazorExcludeRefAssembliesFromPublish>false</MvcRazorExcludeRefAssembliesFromPublish> 
 <MvcRazorCompileOnPublish>false</MvcRazorCompileOnPublish>
  </PropertyGroup>
```

using Project Sdk="Microsoft.NET.Sdk.Web> cause Project Sdk="Microsoft.NET.Sdk.Razor" missing the Publishing templates for azure and iis.

cshtml page should be dynamic compiled razore page on first request or filesystem watcher changed event

complete callstack http://coredemo.ppedv.de/tinyPage

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us, @hannespreishuber.
@pranavkm, can you please look into this? Thanks!

**Comment by pranavkm** [comment]
The flags look correct. Can you tell if the `refs` folder is part of the publish output? If it's not, try running `dotnet publish /bl` and attach the generated `msbuild.binlog

**Comment by hannespreishuber** [comment]
there is no refs folder on published Website (done with Visual Studio)


[msbuild.zip](https://github.com/aspnet/AspNetCore/files/2856342/msbuild.zip)

**Comment by pranavkm** [comment]
At least when you do a local publish, the build log says that publishing produces a refs folder.

![image](https://user-images.githubusercontent.com/174281/52645271-4db19800-2e95-11e9-8802-84fe1abea69c.png)

Perhaps try doing a clean publish to see if that works? If you continue having trouble with it, I'd recommend using the VS feedback tool to report an issue. In the meanwhile, you could FTP copy the locally produced refs folder to the server to unblock yourself.

**Comment by hannespreishuber** [comment]
I have done the repro several times, delete all files, publish all, create new Razor page, publish page
there are even no BIN or debug Folders. All DLLs are in root of Website?
so what profile is dotnet publish /bl using?

**Comment by pranavkm** [comment]
@vijayrkn would you be able to help here?

**Comment by hannespreishuber** [comment]
ok, found the local bin\debug…\publis Folder and copied the refs Folder to Website  - and now it works 
magic-
Bug in Visual Studio?

**Comment by pranavkm** [comment]
> Bug in Visual Studio?

I'd recommend filing a VS feedback issue so that it can be tracked by the right folks. There isn't a Razor SDK \ runtime issue here.

> btw where in docu is something written about the refs Folder?

I am not aware of any documentation for Microsoft.NET.Sdk which is responsible for producing the refs directory.

**Comment by IanGleeson** [comment]
I'm having an issue with this as well.

I've published my Net Core 2.2 application to Azure App Service and receive this message "Cannot find compilation library location for package 'Microsoft.NETCore.App'"

I then add the ref folder as mentioned above and it works again. The issue I have is that it is much slower now than it used to be. Looking at Chrome Inspector the TimeToFirstByte for some pages can go over 15 seconds. Could this solution cause this? If so, does anyone know a fix?

**Comment by pranavkm** [comment]
**Hi, it looks like you are posting on a closed issue/PR/commit!**

We're very likely to lose track of your bug/feedback/question unless you:
1. Open a new issue
2. Explain very clearly what you need help with
3. If you think you have found a bug, include detailed repro steps so that we can investigate the problem

Thanks!

**Comment by IanGleeson** [comment]
Cheers, I'll do that.

**Comment by jjxtra** [comment]
Also seeing this in self contained Windows builds... shouldn't they not require any part of the dotnet core framework installed?
